twill-yarn/src/main/java/org/apache/twill/internal/container/TwillContainerService.java

```diff
@@ -56,24 +66,27 @@
 
   public TwillContainerService(BasicTwillContext context, ContainerInfo containerInfo, ZKClient zkClient,
                                RunId runId, TwillRunnableSpecification specification, ClassLoader classLoader,
-                               Location applicationLocation) {
+                               Location applicationLocation, Map<String, LogEntry.Level> logLevels) {
     super(zkClient, runId, applicationLocation);
 
     this.specification = specification;
     this.classLoader = classLoader;
-    this.containerLiveNodeData = createLiveNodeData(containerInfo);
+    this.containerLiveNodeData = createLiveNodeData(containerInfo,
+                                                    isLoggerContext()
+                                                      ? logLevels : Collections.<String, LogEntry.Level>emptyMap());
     this.context = context;
   }
 
-  private ContainerLiveNodeData createLiveNodeData(ContainerInfo containerInfo) {
+  private ContainerLiveNodeData createLiveNodeData(ContainerInfo containerInfo,
+                                                   Map<String, LogEntry.Level> logLevels) {
     // if debugging is enabled, log the port and register it in service discovery.
     String debugPort = System.getProperty("twill.debug.port");
     if (debugPort != null) {
       LOG.info("JVM is listening for debugger on port {}", debugPort);
     }
     return new ContainerLiveNodeData(containerInfo.getId(),
                                      containerInfo.getHost().getCanonicalHostName(),
-                                     debugPort);
+                                     debugPort, logLevels);
```

Comment:
I think we should filter out the one that has `null` value, since those are representing reset.